### DISCUSSION TITLE
Documentation on everything rendering

### DISCUSSION THREAD

**Discussion by KirmesBude** [bevyengine/bevy#14617] [Q&A]
I am looking for documentation or tutorials for rendering in bevy. Some of those things might not be strictly bevy specific, but would be nice to know how they are used here or how they work.
What are RenderPasses, Pipelines, Pipeline specalization, RenderCommands and how do they relate to each other/what do they do?
What is supposed to happen in extract, prepare and queue?

In specific I want to try to understand every part on the simplest level to e.g. draw a triangle (entity with marker component and transform), which I can extend to learn more advanced stuff (draw texture, draw mesh, ...).

**Comment by bugsweeper** [comment]
Many things in bevy based on wgpu (bevy is just paralellized automatization over it), so you can try to go through it's [Learn article](https://sotrh.github.io/learn-wgpu/). It will show step-by-step what are pipelines and render commands for example

**Reply by KirmesBude** [reply]
That is a really nice article. But already in the first steps it gets very complicated, because to do the same stuff (RenderPass) you need to interface with bevy render graphs, which do not seem to have extended documentation in the book or relevant examples.
Guess I have to play around with it a bit.

**Comment by jf908** [comment]
There is some WIP documentation on this which could help you

https://hackmd.io/@bevy/rendering_summary from [this PR](https://github.com/bevyengine/bevy-website/pull/1080)
